feat: config cleanup finalize

takb opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

takb committed:
feat: config @ arg > env > default locations, config info messages, default disabled profiles

MichaelsJP committed:
feat(config): Add bash script to rewrite the application.yml to the ors-config.yml

This feature intents to keep the application.yml and the ors-config.yml in sync. The ors-config.yml is intented as a minimal working configuration for users that want an up and running system without configuration. The in-depth configs are still in the ors-config.yml but commented out. The sync script will make sure in a workflow that if the original application.yml changes, the ors-config.yml mirrors the new changes but commented out.

MichaelsJP committed:
feat(config): Add workflow for the automatic config conversion

The workflow is triggered on PRs for main and releases/** branches if the PR is set for review or synchronized.

sfendrich commented on .github/utils/config_rewrite.sh:
```diff
@@ -0,0 +1,30 @@
+sed -i 's/#    profiles:/    profiles:/g' "$output_file"
+
+# Replace the default_profile. Ignore the value of enabled and always set to false.
+awk -i inplace '/#    profile_default:/{getline; if ($0 ~ /#      enabled:/) {print "    profile_default:"; print "      enabled: false"; next}} {print}' "$output_file"
```
Line 23 can be removed.

MichaelsJP replied:
gone

sfendrich commented on .github/utils/config_rewrite.sh:
```diff
@@ -0,0 +1,30 @@
+awk -i inplace '/#    profile_default:/{getline; if ($0 ~ /#      enabled:/) {print "    profile_default:"; print "      enabled: false"; next}} {print}' "$output_file"
+
+# Replace the individual profiles. Ignore the value of enabled and always set to false.
+for profile in car hgv bike-regular bike-mountain bike-road bike-electric walking hiking public-transport; do
```
Here, all profiles should be left untouched, except car-profile should be turned `enabled: true`.

MichaelsJP replied:
Fixed

sfendrich commented on .github/utils/config_rewrite.sh:
```diff
@@ -0,0 +1,30 @@
+sed -i 's/#ors:/ors:/g' "$output_file"
+# Replace '#  engine:' with '  engine:'
+sed -i 's/#  engine:/  engine:/g' "$output_file"
+# Replace '#    source_file:' with '    source_file: "YOUR_FILE"'
```
Should we use heidelberg instead of "YOUR_FILE"?

MichaelsJP replied:
It now has the default Heidelberg pbf with the path from the root folder of the repo. It allows us to immediately  start ors with spring without any additional configuration.

sfendrich commented on .github/utils/config_rewrite.sh:
```diff
@@ -0,0 +1,30 @@
+# Replace '#  engine:' with '  engine:'
+sed -i 's/#  engine:/  engine:/g' "$output_file"
+# Replace '#    source_file:' with '    source_file: "YOUR_FILE"'
+sed -i 's/#    source_file:/    source_file: "YOUR_FILE"/g' "$output_file"
```
This might be risky, if other options also include a field called `source_file

MichaelsJP replied:
I moved the source_file parameter directly under engine: so the script can parse the two together. Makes the possibility of namespace clashes less likely.

sfendrich commented on ors-api/ors-config.yml:
```diff
@@ -1,6 +1,6 @@
 ##### openrouteservice settings file #####
 # This file contains parameters for openrouteservice.
-# For a full list of possible parameters see documentation, below are some examples.
+# For a full list of possible parameters see https://giscience.github.io/openrouteservice/, below are some examples.
```
This will be overwritten and should be placed in application.yml

takb replied:
done.

sfendrich commented on ors-api/src/main/java/org/heigit/ors/api/ORSEnvironmentPostProcessor.java:
```diff
@@ -22,23 +22,34 @@ public class ORSEnvironmentPostProcessor implements EnvironmentPostProcessor {
+            } catch (IllegalStateException | IOException ignored) {
             }
         }
+        System.setProperty(ORS_CONFIG_LOCATION_PROPERTY, output);
```
A deferred log would be more elegant: https://stackoverflow.com/questions/42839798/how-to-log-errors-in-a-environmentpostprocessor-execution

sfendrich commented on ors-api/src/main/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListener.java:
```diff
@@ -46,23 +47,25 @@ public ORSInitContextListener(EngineProperties engineProperties) {
-            }
-            if (!StringUtility.isNullOrEmpty(System.getProperty(ORS_CONFIG_LOCATION_PROPERTY))) {
-                LOGGER.debug("Configuration loaded by ARG, location: " + System.getProperty(ORS_CONFIG_LOCATION_PROPERTY));
+        LOGGER.info("");
```
The following is related to the deferred log ...

sfendrich reviewed (changes requested):
A few minor issues.

MichaelsJP committed:
chore(config): automatic conversion of application.yml to ors-config.yml

takb committed:
fix: logging with DeferredLogger

MichaelsJP committed:
feat(config): Add workflow for the automatic config conversion (#1656)

takb committed:
feat: sort ENV list

takb committed:
Merge branch 'main' into feat/config-cleanup-finalize

takb committed:
fix: api tests, default source file, ors-config.yml location

MichaelsJP committed:
fix(config): Adjust the conversion script

Some prior changes got overwritten. This introduces the old state as well as an additional testing if the config is actually in the right format to be parsed.

MichaelsJP committed:
fix(config): Move the source_file parameter for conversion reasons

To allow a smooth conversion of the application.yml the source_file parameter needs to be next after engine:. Else, awk operations will just become unnecessary tedious.

MichaelsJP committed:
fix(config): Adjust the conversion script (#1660)

takb committed:
fix: clean exit on graphs_data_access misconfiguration

sfendrich reviewed (approved):
LGTM! Thank you.

MichaelsJP committed:
fix(docker): Fix the dockerfile to use the example .yml config properly

The old dockerfile was coined to use the json config. This commit just makes sure everything works again. The whole setup will probably get a revamp before release anyways

MichaelsJP committed:
chore(docker): Let the docker builds run on every PR not just to main

takb committed:
fix(dockerfile): Repair the docker setup so it can use the yml config properly (#1663)
